src/processor/graph-update-queue-consumer.service.ts

```diff
@@ -85,10 +113,40 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
     }
 
     try {
-      await this.graphSdkService.updateUserGraph(job.id ?? '', job.data.dsnpId, job.data.providerId, job.data.processTransitiveUpdates);
-      this.logger.verbose(`Successfully completed job ${job.id}`);
+      let doTrack = false;
+      if (await this.cacheManager.doesJobKeyExist(job.id!)) {
```

Comment:
Do we want to throw an error if the job id is undefined?
```suggestion
      if (job.id && await this.cacheManager.doesJobKeyExist(job.id)) {
```

Reply:
This is odd. I don't think it's possible for the queue to be consuming a job without an ID... not sure why the interface specifies that field as optional, when I can't see how it could ever be missing. I think maybe it's for easily working with Jobs on the producer side of the queue.